spec/fixtures/inherited_json_schema.json

```diff
@@ -0,0 +1,78 @@
+{
+  "schema": {
```

Comment:
To comply with JSON schema standard, I think the result of the parser should be:

```json
{
  "schema": {
    "allOf": [
      {
        "schema": null,
        "type": "object",
        "required": [
          "personal_details",
          "login_details",
          "agreements"
        ],
        "properties": {
          "personal_details": {
            "type": "object",
            "properties": {
              "first_name": {
                "type": "string"
              },
              "last_name": {
                "type": "string"
              },
              "birthdate": {
                "type": "string",
                "format": "date"
              },
              "email": {
                "type": "string",
                "format": "email"
              },
              "email_confirmation": {
                "type": "string",
                "format": "email"
              }
            },
            "required": [
              "first_name",
              "last_name",
              "birthdate",
              "email",
              "email_confirmation"
            ]
          }
        },
        "login_details": {
          "type": "object",
          "properties": {
            "star_alliance_id": {
              "type": "string",
              "pattern": "^[0-9]{10}$",
              "format": "id"
            }
          },
          "required": [
            "star_alliance_id"
          ]
        },
        "agreements": {
          "type": "object",
          "required": [
            "terms"
          ],
          "properties": null,
          "terms": {
            "type": "boolean",
            "x-display-type": "checkbox"
          }
        }
      },
      {
        "type": "object",
        "properties": {
          "login_details": {
            "type": "object",
            "required": [
              "star_alliance_id",
              "password",
              "password_confirmation"
            ],
            "properties": {
              "star_alliance_id": {
                "type": "string",
                "pattern": "^[0-9]{10}$",
                "format": "id"
              },
              "password": {
                "type": "string",
                "format": "password"
              },
              "password_confirmation": {
                "type": "string",
                "format": "password"
              }
            }
          }
        }
      }
    ]
  }
}
```

Reply:
You can check whether the form can be rendered correctly in https://rjsf-team.github.io/react-jsonschema-form/